[lldb/Swift] Add progress report callback when loading a module

medismailben opened the pull request:
This patch introduces a new `ReportModuleLoadingProgress` callback that
gets invoked by the Swift AST Context when loading new modules and
module overlays.

This patch also updates `TestSwiftProgressReporting` to check that these
new progress reports are broadcasted correctly.

rdar://94165195

Signed-off-by: Med Ismail Bennani <medismail.bennani@gmail.com>

medismailben commented:
This requires [apple:swift/59181](https://github.com/apple/swift/pull/59181)

medismailben commented:
@swift-ci test

adrian-prantl commented on lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp:
```diff
@@ -3138,6 +3138,7 @@ swift::ASTContext *SwiftASTContext::GetASTContext() {
       GetLanguageOptions(), GetTypeCheckerOptions(), GetSILOptions(),
       GetSearchPathOptions(), GetClangImporterOptions(),
       GetSymbolGraphOptions(), GetSourceManager(), GetDiagnosticEngine()));
```
Why didn't we add this as a default argument to the constructor?

adrian-prantl commented on lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp:
```diff
@@ -3407,6 +3408,15 @@ void SwiftASTContext::CacheModule(swift::ModuleDecl *module) {
 
+bool SwiftASTContext::ReportModuleLoadingProgress(llvm::StringRef module_name,
+                                                  bool is_overlay) {
+  Progress progress(llvm::formatv(is_overlay ? "Loading overlay module {0}"
```
I think "Importing" would make more sense, since the statement in the language is also `import`.

medismailben commented:
@swift-ci test

medismailben commented:
@swift-ci test

medismailben commented:
@swift-ci test

medismailben committed:
[lldb/Swift] Add progress report callback when loading a module

This patch introduces a new ReportModuleLoadingProgress callback that
gets invoked by the Swift AST Context when loading new modules and
module overlays.

This patch also updates TestSwiftProgressReporting to check that these
new progress reports are broadcasted correctly.

rdar://94165195

Signed-off-by: Med Ismail Bennani <medismail.bennani@gmail.com>

medismailben commented:
@swift-ci test
